### ISSUE TITLE
Training Loss Unchanged During Fine-Tuning when Varying Negative Pairs

### ISSUE THREAD

**Issue by doramasma** [FlagOpen/FlagEmbedding#958]
Hello! I'm experiencing an issue or unexpected behavior during the fine-tuning process where the training loss remains unchanged, despite varying the number of negative pairs in the dataset used for fine-tuning. Below are the details:

I've been performing different fine-tuning experiments using datasets composed of queries, where each query has 1 positive pair (the positive pair associated with the query remains the same across the three datasets for a given query). The difference between the datasets is the varying number of negative pairs. I have created three datasets:

  - Dataset 1: 1 negative pair per query
  - Dataset 2: 4 negative pairs per query
  - Dataset 3: 10 negative pairs per query

**Expected Behavior**:
I expected the training loss to change as the number of negative pairs increases in the datasets. I'm not saying that should be better or worse, just change the loss, as the model should be trained with different information, right? 

**Observed Behavior**:
The training loss does not exhibit any noticeable change despite the variation in the number of negative pairs. As you can see in the following image, I've integrated the logs with Weights and Biases, and the loss appears to be the same.

<img width="1429" alt="image" src="https://github.com/FlagOpen/FlagEmbedding/assets/7035260/21df7cc0-5904-4809-89ad-d0ec48a986e4">

Any ideas or advice? Do I need to activate a flag or configure something special to use the list of pairs during the fine-tuning? 

```json
{
  "query": "string",
  "pos": ["list of positive strings"],
  "neg": ["list of negative strings"]
}
```

Thanks in advance!

**Comment by staoxiao** [comment]
@doramasma , do you change the hyper-parameter `--train_group_size` (which can control the number of used negatives)?

**Comment by staoxiao** [comment]
@doramasma , during the training, we will sample one positive example from the list `'pos'`, and then sample `train_group_size - 1` negatives from list `'neg'`.  So, if you have 5 positive and 5 negative pairs, you can set `train_group_size` <= 6.

**Comment by doramasma** [comment]
Thanks a lot!
